astropy/coordinates/angles.py

```diff
@@ -124,7 +125,8 @@ def __new__(cls, angle, unit=None, dtype=None, copy=True, **kwargs):
                     angle_unit = unit
 
                 if isinstance(angle, tuple):
-                    angle = cls._tuple_to_float(angle, angle_unit)
+                    angle = cls._tuple_to_float(angle, angle_unit,
```

Comment:
This suppression of the deprecation seems weird. Wouldn't it be better to just have a direct implementation that does the right thing? That way, once the deprecation period is over, we can just delete `dms_to_degrees` and `_tuple_to_float`. 

Here, because the parser has already ensured the various elements are of the right type, I think it is as simple as 
```
b, m, s = angle
if angle_unit == u.hr:
    form._check_hour_range(b)
form._check_minute_range(m)
form._check_second_range(s)
angle = np.copysign(np.abs(b]) + m / 60. + s / 3600., b) << angle_unit
```

Reply:
Fair point - I was trying to do the minimal possible, but this is a better idea!

Reply:
Hmm.  On second thought, when I went to implement this I got a big piles of errors.  It's partly because the tuples are not constrained to be length-3, but even when I account for that something else is broken.

So I think we should make it a project for 5.2 to do this correctly, but not have it hold up this PR which is already going to need special permission to make it into 5.1!